### ISSUE TITLE
Support passing a TypeScript enum to `enum` validator in schema

### ISSUE THREAD

**Issue by vkarpov15** [Automattic/mongoose#9546]
Right now you need to use `Object.values()` to get TypeScript enums to work with the `enum` validator:

```typescript
const promotionSchema = new Schema({
   user: { type: Schema.Types.ObjectId, ref: 'User' },
   product: { type: Schema.Types.ObjectId, ref: 'Product' },
   transactionType: {
      type: String,
      default: ETransactionType.Buy,
      enum: Object.values(ETransactionType)
   }
});
```

Would be nice if Mongoose handled the `Object.values()` part automatically and treated `{ foo: 'FOO', bar: 'BAR' }` as equivalent to `['FOO', 'BAR']` for enums.

**Comment by AbdelrahmanHafez** [comment]
Original discussion: #9535

**Comment by mmykhaylov** [comment]
Typescript still complains for some reason

```
export enum Specialisation {
  Beards = "BEARDS",
  Haircuts = "HAIRCUTS"
}

export interface Barber {
  email: string,
  name: {
    first: string,
    last: string,
  },
  profileImageURL: string,
  specialisation: Specialisation,
  completed?: boolean,
}

interface BarberBaseDocument extends Barber, Document {
  fullName: string
}

export interface BarberDocument extends BarberBaseDocument {
  appointmentIDS: Types.Array<Types.ObjectId>;
}

const BarberSchema: Schema<BarberDocument, Model<BarberDocument>> = new Schema({
  email: { type: String, unique: true, required: true },
  name: {
    first: {type: String, required: true},
    last: {type: String, required: true},
  },
  profileImageURL: { type: String, unique: true, },
  specialisation: { type: String, enum: Specialisation, required: true },
  completed: { type: Boolean, default: false },
  appointmentIDS: [
    {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'Appointment',
    },
  ],
});
```

In BarberSchema it complains on the specialisation field saying
```
Type '{ type: StringConstructor; enum: typeof Specialisation; required: true; }' is not assignable to type 'SchemaDefinitionProperty<undefined>'.
Type '{ type: StringConstructor; enum: typeof Specialisation; required: true; }' is not assignable to type 'string'.
```

Where is the mistake?

**Comment by vkarpov15** [comment]
@mmykhaylov we looked into this, it looks like TypeScript only complains if you set `required: true` and turn on strict mode. We added a fix that will be released in v5.11.14 :+1:
